sources/HUBViewControllerImplementation.m

```diff
@@ -47,10 +47,11 @@
 #import "HUBActionRegistry.h"
 #import "HUBActionHandler.h"
 #import "HUBViewModelDiff.h"
+#import "HUBComponentGestureRecognizer.h"
 
 NS_ASSUME_NONNULL_BEGIN
 
-@interface HUBViewControllerImplementation () <HUBViewModelLoaderDelegate, HUBImageLoaderDelegate, HUBComponentWrapperDelegate, UICollectionViewDataSource, UICollectionViewDelegate>
+@interface HUBViewControllerImplementation () <HUBViewModelLoaderDelegate, HUBImageLoaderDelegate, HUBComponentWrapperDelegate, UICollectionViewDataSource, UICollectionViewDelegate, UIGestureRecognizerDelegate>
```

Comment:
Totally! We should add a GitHub issue for it with subtasks. There's a few "quick wins", like moving image loading out of it.